CDCgov/prime-simplereport
# Johanna/5847 manage facility

# FRONTEND/BACKEND PULL REQUEST

## Related Issue

- Resolves #5847 

## Changes Proposed
- Create page manage facility
- Add frontend unit testing
- Create new endpoint `facilityStats` to retrieve the number of users and patients assigned only to the facility.
- Modify endpoint `markFacilityAsDeleted` to remove in progress tests in the test queue if the facility is being marked as deleted.
- Added backend unit test
- Added e2e testing and a11y checks.

## Additional Information
- Jayna and Laura Kerry have approved of the content.
- I also fixed the modal component to make the `x` button available as it was getting hidden behind the modal content and was not possible to click it with the mouse.

## Testing
Code is deployed in [dev](https://dev.simplereport.gov/app/admin/manage-facility)

- Verify that you can delete a facility and this is reflected in the database.
- Verify that if there is a test being conducted while you delete the facility the test gets removed and the application does not crash.
[Manage facility- Delete facility demo video](https://github.com/CDCgov/prime-simplereport/assets/103958711/664b9e9c-a944-46dd-8e66-86df4d09f8e4)
![Screenshot 2023-08-14 at 10 49 01 AM](https://github.com/CDCgov/prime-simplereport/assets/103958711/fd8dcc16-0d32-4d8e-b9ac-1c53358f928c)
![Screenshot 2023-08-14 at 10 49 09 AM](https://github.com/CDCgov/prime-simplereport/assets/103958711/771c43ae-6128-4129-ad72-f893c08b24bc)
![Screenshot 2023-08-14 at 10 49 15 AM](https://github.com/CDCgov/prime-simplereport/assets/103958711/214a37e3-7094-433e-b777-03d8467ba27e)
